server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -298,6 +298,20 @@ public DiscoveryNode findByAddress(TransportAddress address) {
         return null;
     }
 
+    /**
+     * Get a node by its name
+     *
+     * @return node identified by the given name or <code>null</code> if no such node exists
+     */
+    public DiscoveryNode findByName(String name) {
```

Comment:
The result here relies on order of the map in case of duplicates, which seems to indicate that the usefulness of sharing the method here is smal. But we are not really concerned about the node itself, only whether it exists. I think we should instead have a method `boolean hasByName(String name)`.